src/firefly/js/charts/ui/Histogram.jsx

```diff
@@ -0,0 +1,389 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+import React, {PropTypes} from 'react';
+import ReactHighcharts from 'react-highcharts/bundle/highcharts';
+import numeral from 'numeral';
+import {getFormatString} from '../../util/MathUtil.js';
+import {logError} from '../../util/WebUtil.js';
+
+/*
+ * @param {String} color - hex color, exactly seven characters log, starting with '#'
+ * @param {Number} persentage (0.1 means 10 percent lighter, -0.1 - 10 percent darker)
+ * @return {String} lighter or darker shade of the given hex color
+ * from http://stackoverflow.com/questions/5560248/programmatically-lighten-or-darken-a-hex-color-or-rgb-and-blend-colors
+ */
+function shadeColor(color, percent) {
+    var f=parseInt(color.slice(1),16),t=percent<0?0:255,p=percent<0?percent*-1:percent,R=f>>16,G=f>>8&0x00FF,B=f&0x0000FF;
+    return `#${(0x1000000+(Math.round((t-R)*p)+R)*0x10000+(Math.round((t-G)*p)+G)*0x100+(Math.round((t-B)*p)+B)).toString(16).slice(1)}`;
+}
+
+
+export class Histogram extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.setChartConfig = this.setChartConfig.bind(this);
+        this.validateData = this.validateData.bind(this);
+    }
+
+    shouldComponentUpdate(nextProps) {
+        const {data, width, height, logs, reversed, desc, binColor} = this.props;
+        // should rerender only if data or bin color has changed
+        // otherwise just change the existing chart
+        if (data != nextProps.data || binColor !== nextProps.binColor) { return true; }
+        const chart = this.refs.chart && this.refs.chart.getChart();
+        if (chart) {
+            let doUpdate = false;
+            if (height !== nextProps.height || width !== nextProps.width ) {
+                chart.setSize(nextProps.width, nextProps.height, false);
+                return false;
+            }
+
+            if (desc !== nextProps.desc) {
+                chart.xAxis[0].setTitle(nextProps.desc, false);
+                doUpdate = true;
+            }
+            if (reversed !== nextProps.reversed){
+                const nreversed = nextProps.reversed;
+                const yReversed = (nreversed && nreversed.indexOf('y')>-1 ? true : false);
+                const xReversed = (nreversed && nreversed.indexOf('x')>-1 ? true : false);
+                chart.xAxis[0].update({'reversed' : xReversed}, false);
+                chart.yAxis[0].update({'reversed' : yReversed}, false);
+                doUpdate = true;
+            }
+            const nreversed = nextProps.reversed;
+            if (reversed !== nreversed){
+                const yReversed = (nreversed && nreversed.indexOf('y')>-1 ? true : false);
+                const xReversed = (nreversed && nreversed.indexOf('x')>-1 ? true : false);
```

Comment:
alternative.. not necessarily better:  !!(nreversed && nreversed.indexOf('x')>-1)